src/easier68k/core/opcodes/dc.py

```diff
@@ -0,0 +1,392 @@
+                    if current_param:
+                        if not is_string:  # Try parsing this value for a literal (if it's a string it's almost certainly fine)
+                            assert parse_literal(current_param.strip()) is not None, 'Error parsing literal'
+                        param_count += 1
+
+                    is_string = False
+                    quote_delim = None
+                    current_param = ''
+                    continue
+
+                if quote_delim and c == quote_delim:  # Breaking out of the quote
+                    quote_delim = None
+                    continue
+
+                if c == '\'':
+                    if not quote_delim:  # This is the start of a quote
+                        quote_delim = '\''
+                        assert not current_param.strip(), "Can't have two quotes in a row in one parameter"
+                        current_param = ''
+                        is_string = True
+                        continue
+
+                if c == '\"':
+                    if not quote_delim:  # This is the start of a quote
+                        quote_delim = '\"'
+                        assert not current_param.strip(), "Can't have two quotes in a row in one parameter"
+                        current_param = ''
+                        is_string = True
+                        continue
+
+                current_param += c
+
+            assert not quote_delim, 'Unclosed quotes'
+            if current_param:
+                if not is_string:  # Try parsing this value for a literal (if it's a string it's almost certainly fine)
+                    assert parse_literal(current_param.strip()) is not None, 'Error parsing literal'
+                param_count += 1
+
+            assert param_count > 0, 'Must have at least one parameter'
+
+        except AssertionError as e:
+            issues.append((e.args[0], 'ERROR'))
+            return False, issues
+
+        return True, issues
+
+    @staticmethod
+    def get_word_length(command: str, parameters: str) -> (int, list):
+        """
+        Gets the final length of a command in memory in words
+        NOTE: for DC.B, this will round UP to make it a full word, even though it won't use the full memory
+
+        >>> DC.get_word_length('DC.B', '$0A, $0B')[0]
+        1
+
+        >>> DC.get_word_length('DC.W', '$0A, $0B')[0]
+        2
+
+        >>> DC.get_word_length('DC.L', '$0A, $0B')[0]
+        4
+
+        >>> DC.get_word_length('DC.B', '\\'Hai!\\'')[0]
+        2
+
+        >>> DC.get_word_length('DC.W', '\\'Hai!\\'')[0]
+        2
+
+        >>> DC.get_word_length('DC.L', '\\'Hai!\\'')[0]
+        4
+
+        :param command: The command itself (e.g. 'MOVE.B', 'LEA', etc.)
+        :param parameters:  The parameters after the command (such as the source and destination of a move)
+        :return: The length of the command in memory (in words) and a list of issues/warnings encountered during assembly
```

Comment:
Why measure length in words and not length in bytes? I'm fairly certain elsewhere measures in bytes.
This would also get around the issues of adding 1/2 for byte values.

Reply:
I had some reason for it, I think it's mainly because instructions never come out to byte-level lengths (they're always an even word length): I add half here because it gets rounded up in the end. It would work either way, so I'm up for changing it, it just has to remain consistent for all opcodes.

Reply:
Could multiply the result to return bytes so that it can be used elsewhere.